cub/test/c2h/custom_type.cuh

```diff
@@ -131,6 +131,11 @@ public:
   {
     return lhs.key == rhs.key && lhs.val == rhs.val;
   }
+
+  friend __host__ __device__ bool operator!=(const CustomType& lhs, const CustomType& rhs)
```

Comment:
suggestion: the motivation behind policy is to check if a given algorithm relies on documented set of type requirements. If we promise to only use `==` but accidentally rely on `!=` it'll be a problem. If you do need `!=`, maybe it's worth extracting into a separate policy?

Reply:
If a type promises to be equality comparable, it should expose both, `==` and `!=`. This was even built into C++20, where the compiler can rewrite any expression `a != b` as `!(a == b)` if no `operator!=` is found. I strongly believe we should provide both operators for the policy. Also, any type that only defines `operator==` but not `operator!=` is broken.